### PR TITLE
Fix ast view and command registration

### PR DESCRIPTION
**Pull request by aeisenberg**
Two small bugs:

1. The AST view command was viewing the wrong ast when the command was
   selected from the context menu. It was always selecting the active
   editor instead of the item selected in the file menu.
2. The `codeql.showLogs` command was not being registered properly.
   With this change, there is uniform error handling, telemetry,
   and disposal.

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [x] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 - Fix a bug where the results view moved column even when it was already visible. [#1070](https://github.com/github/vscode-codeql/pull/1070)
 - Add packaging-related commands. _CodeQL: Download Packs_ downloads query packs from the package registry that can be run locally, and _CodeQL: Install Pack Dependencies_ installs dependencies for packs in your workspace. [#1076](https://github.com/github/vscode-codeql/pull/1076)
 - Add query history items as soon as a query is run, including new icons for each history item. [#1094](https://github.com/github/vscode-codeql/pull/1094)
+- Fix a bug where invoking _View AST_ from the file explorer would not view the selected file. Instead it would view the active editor. [#1113](https://github.com/github/vscode-codeql/pull/1113)
```
Comment (shati-patel) [inline extensions/ql-vscode/CHANGELOG.md:10]: Not yet released! 😸 Can you move this higher up to `[UNRELEASED]`? 🚀

**Review by shati-patel — COMMENTED** [review]
Looks good! 🙌🏽 

Just a minor comment/question: I accidentally selected "View AST" on a folder, which then showed up in the AST view (but doesn't have anything useful) 😅 Should this throw the error `Cannot view the AST. Please select a valid source file...`?

755FEBDC-F392-4787-9444-68869B795EA9

File: extensions/ql-vscode/CHANGELOG.md (hunk 1 above)
Comment (aeisenberg) [reply extensions/ql-vscode/CHANGELOG.md:10]: That's from a bad merge. I'll fix.

**Comment by aeisenberg** [comment]
Thanks for finding that bug. I hadn't even thought of trying to run on a directory.

**Review by shati-patel — APPROVED** [review]
🎉 Thanks for fixing this! 💖

**Pull request merged**